reactivesocket-core/src/main/java/io/reactivesocket/ServerReactiveSocket.java

```diff
@@ -321,33 +330,31 @@ private synchronized void cleanUpReceivingSubscription(Subscription subscription
     }
 
     private Mono<Void> handleChannel(int streamId, Frame firstFrame) {
-        return Mono.defer(() -> {
-            UnicastProcessor<Frame> frames = UnicastProcessor.create();
-            Flux<Payload> payloads = frames
-                .doOnCancel(() -> {
-                    if (connection.availability() > 0.0) {
-                        connection.sendOne(Frame.Cancel.from(streamId)).subscribe(null, errorConsumer::accept);
-                    }
-                })
-                .doOnError(t -> {
-                    if (connection.availability() > 0.0) {
-                        connection.sendOne(Frame.Error.from(streamId, t)).subscribe(null, errorConsumer::accept);
-                    }
-                })
-                .doOnRequest(l -> {
-                    if (connection.availability() > 0.0) {
-                        connection.sendOne(Frame.RequestN.from(streamId, l)).subscribe(null, errorConsumer::accept);
-                    }
-                })
-                .cast(Payload.class);
+        UnicastProcessor<Frame> frames = UnicastProcessor.create();
+        Flux<Payload> payloads = frames
+            .doOnCancel(() -> {
+                if (connection.availability() > 0.0) {
```

Comment:
tangential to this review.  Wouldn't this check be better handled within the connection?  it could always change between the check and the call anyway.